Go/Go.sublime-syntax

```diff
@@ -432,28 +449,59 @@ contexts:
     # Dot without integer
     - match: \.\d+(?:{{exponent}}\d+)?
       scope: invalid.deprecated.go
+    # Hex float literal, no fraction
+    - match: '(0[xX]){{hdigits}}([pP])([+-])?{{ddigits}}(i)?'
+      scope: constant.numeric.float.go
+      captures:
+        1: punctuation.definition.numeric.hexadecimal.go
+        2: punctuation.section.exponent.go
+        3: keyword.operator.go
```

Comment:
How about removing the special scoping for the exponent (`E+`, `p+`, ...) which is used in this syntax (`punctuation.separator.exponent` in some places and `punctuation.section.exponent` for this example), because it seems to not be used elsewhere (there is a PR #2166 to remove `punctuation.definition.exponent` in Erlang)? In my opinion it doesn't add much value and doesn't really look well in color schemes that target `punctuation.separator`. Also `keyword.operator` for the `+`/`-` doesn't make sense here in my opinion.

Reply:
Good idea. Looking at the other syntaxes it seems most don't have extra scopes for the exponent and/or `+`/`-` sections of numeric literals. Removed in f894f19, 1947eca, and 8c2b119.